etl/extract.py

```diff
@@ -0,0 +1,44 @@
+import re
+from datetime import datetime
+from pathlib import Path
+
+import pandas as pd
+
+
+def extract_fund_info(filename: str) -> tuple[str | None, str | None]:
+    line = filename
+    line = re.sub(
+        r"^(mend-report|Fund|Report-of|rpt|TT_monthly)", "", line, flags=re.IGNORECASE
+    )
+    line = line.strip(" -_")
+
+    name_match = re.match(r"([A-Za-z\s]+?)(?=[._])", line)
+    fund_name = name_match.group(1).strip() if name_match else None
+
+    date_match = re.search(
+        r"(\d{2}[-_]\d{2}[-_]\d{4}|\d{8}|\d{4}[-_]\d{2}[-_]\d{2})", line
```

Comment:
For the files given, a much simpler regexp may be used that just extracts as many groups of digits as it finds.